modules/imgcodecs/perf/perf_main.cpp

```diff
@@ -8,3 +8,35 @@
 #endif
 
 CV_PERF_TEST_MAIN(imgcodecs)
+
+namespace opencv_test
+{
+
+using namespace perf;
+
+#define TYPICAL_MAT_SIZES_TIFF_DECODE  cv::Size(128, 128), cv::Size(256, 256), cv::Size(512, 512), cv::Size(1024, 1024)
+#define TYPICAL_MAT_TYPES_TIFF_DECODE  CV_8UC1, CV_8UC3, CV_16UC1, CV_16UC3
+#define TYPICAL_MATS_TIFF_DECODE       testing::Combine( testing::Values( TYPICAL_MAT_SIZES_TIFF_DECODE), testing::Values( TYPICAL_MAT_TYPES_TIFF_DECODE) )
+
+PERF_TEST_P(Size_MatType, tiffDecode, TYPICAL_MATS_TIFF_DECODE)
```

Comment:
We usually have only `main` and infrastructure related code in `perf_main.cpp`. Please extract test code to new file, let say `perf_tiff.cpp`.

Reply:
:+1: